src/Week6/HW6.hs

```diff
@@ -0,0 +1,120 @@
+{-# OPTIONS_GHC -Wall #-}
+module Week6.HW6 where
+
+import Data.List
+import Data.Functor
+
+-- Exercise 1 -----------------------------------------
+
+-- Write a Fibonacci function
+fib :: Integer -> Integer
+fib 1 = 1
+fib 2 = 1
+fib n = fib (n - 2) + fib (n - 1)
+
+-- Define an infinite Fibonnaci sequence
+fibs1 :: [Integer]
+fibs1 = fib <$> [1..]
+
+-- Exercise 2 -----------------------------------------
+
+-- Define a faster infinite Fibonnaci sequence
+fibs2 :: [Integer]
+fibs2 = 1 : 1 : zipWith (+) fibs2 (tail fibs2)
+
+-- Exercise 3 -----------------------------------------
+
+data Stream a = Cons a (Stream a)
+
+-- Show instance prints the first 20 elements followed by ellipsis
+instance Show a => Show (Stream a) where
+    show s = "[" ++ intercalate ", " (map show $ take 64 $ streamToList s)
```

Comment:
where's `64` coming from?

Reply:
I guess it's forgotten from testing ruler function :smiley: